addons/mass_mailing/models/mailing.py

```diff
@@ -333,23 +348,51 @@ def _compute_render_model(self):
         for mailing in self:
             mailing.render_model = mailing.mailing_model_real
 
+    @api.depends('ab_testing_enabled', 'ab_testing_pc', 'campaign_id')
+    def _compute_ab_testing_description(self):
+        mailing_ab_test = self.filtered('ab_testing_enabled')
+        (self - mailing_ab_test).ab_testing_description = False
+        for mailing in mailing_ab_test:
+            mailing.ab_testing_description = self.env['ir.qweb']._render(
+                'mass_mailing.ab_testing_description',
+                mailing._get_ab_testing_description_values()
+            )
+
     # ------------------------------------------------------
     # ORM
     # ------------------------------------------------------
 
     @api.model_create_multi
     def create(self, vals_list):
         now = fields.Datetime.now()
+        cron = self.env.ref('mass_mailing.ir_cron_mass_mailing_ab_testing').sudo()
         for values in vals_list:
             if values.get('subject') and not values.get('name'):
                 values['name'] = "%s %s" % (values['subject'], now)
             if values.get('body_html'):
                 values['body_html'] = self._convert_inline_images_to_urls(values['body_html'])
-        return super().create(vals_list)
+            if values.get('ab_testing_schedule_datetime'):
+                at = fields.Datetime.from_string(values['ab_testing_schedule_datetime'])
+                cron._trigger(at=at)
+        mailings = super().create(vals_list)
+        campaign_vals = []
+        for mailing in mailings:
+            if mailing.ab_testing_enabled and not mailing.campaign_id:
+                campaign_vals.append(mailing._get_default_ab_testing_campaign_values())
+        self.env['utm.campaign'].create(campaign_vals)
+        return mailings
 
     def write(self, values):
         if values.get('body_html'):
             values['body_html'] = self._convert_inline_images_to_urls(values['body_html'])
+        if values.get('ab_testing_schedule_datetime'):
+            cron = self.env.ref('mass_mailing.ir_cron_mass_mailing_ab_testing').sudo()
+            at = fields.Datetime.from_string(values['ab_testing_schedule_datetime'])
+            cron._trigger(at=at)
+        if values.get('ab_testing_enabled') and not values.get('campaign_id'):
+            values['campaign_id'] = self.env['utm.campaign'].create(self._get_default_ab_testing_campaign_values(values)).id
+        if values.get('campaign_id') is False and self.ab_testing_enabled and 'ab_testing_enabled' not in values:
+            raise ValidationError(_("A campaign should be set when A/B test is enabled"))
```

Comment:
Can this still happen since we create one automagically ?

Reply:
Yes, we create the campaign when ab_testing_enabled is checked. In the case, ab testing is already enabled on a mailing and the user decides to remove the campaign with ab testing still enabled, the ValidationError is raised.

Reply:
Ah, I see, could be nice to add a small comment in the code above that line mentioning this use case.